docs/dev/table/connectors/upsert-kafka.md

```diff
@@ -0,0 +1,183 @@
+---
+title: "Apache Upsert Kafka SQL Connector"
+nav-title: Upsert Kafka
+nav-parent_id: sql-connectors
+nav-pos: 2
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+<span class="label label-primary">Scan Source: Unbounded</span>
+<span class="label label-primary">Sink: Streaming Upsert Mode</span>
+
+* This will be replaced by the TOC
+{:toc}
+
+The Upsert Kafka connector allows for reading data from and writing data into Kafka topics in the upsert fashion.
+
+More precisely, the value part of the record is interpreted as an "UPDATE" of the last value for the same key.
+If the record doesn't exist yet, it will treat the record as an INSERT message. With the tombstone semantic in kafka,
+it will interpret the record whose value part is null as a DELETE message.
+
+
+Dependencies
+------------
+
+To use the connector, add the following Maven dependency to your project:
+{% highlight xml %}
+<dependency>
+  <groupId>org.apache.flink</groupId>
+  <artifactId>flink-connector-kafka_2.11</artifactId>
+  <version>1.12-SNAPSHOT</version>
+</dependency>
+{% endhighlight %}
+
+How to create an Upsert Kafka table
+----------------
+
+The example below shows how to create an Upsert Kafka table:
+
+<div class="codetabs" markdown="1">
+<div data-lang="SQL" markdown="1">
+{% highlight sql %}
+CREATE TABLE upsertKafkaTable (
+  user_id BIGINT,
+  item_id BIGINT,
+  category_id BIGINT,
+  behaviour STRING,
+  ts TIMESTAMP(3),
+  PRIMARY KEY (user_id) NOT ENFORCED
+) WITH (
+  'connector' = 'upsert-kafka',
+  'topic' = 'user_behaviour',
+  'properties.bootstrap.servers' = 'localhost:9092',
+  'key.format' = 'json',
+  'value.format' = 'json',
+  'value.fields-include' = 'ALL'
+)
```

Comment:
Please add a more concreate example for how to insert an aggregation result into this upsert kafka table, and how to read from the upsert kafka table.

You can learn examples in [FLIP-149](https://cwiki.apache.org/confluence/display/FLINK/FLIP-149%3A+Introduce+the+upsert-kafka+Connector#FLIP149:IntroducetheupsertkafkaConnector-Examples) and [this doc](https://docs.ksqldb.io/en/latest/tutorials/examples/).